test/e2e/csv_e2e_test.go

```diff
@@ -4432,89 +4426,86 @@ var csvFailedChecker = buildCSVConditionChecker(operatorsv1alpha1.CSVPhaseFailed
 var csvAnyChecker = buildCSVConditionChecker(operatorsv1alpha1.CSVPhasePending, operatorsv1alpha1.CSVPhaseSucceeded, operatorsv1alpha1.CSVPhaseReplacing, operatorsv1alpha1.CSVPhaseDeleting, operatorsv1alpha1.CSVPhaseFailed)
 var csvCopiedChecker = buildCSVReasonChecker(operatorsv1alpha1.CSVReasonCopied)
 
-func fetchCSV(c versioned.Interface, name, namespace string, checker csvConditionChecker) (*operatorsv1alpha1.ClusterServiceVersion, error) {
-	var fetchedCSV *operatorsv1alpha1.ClusterServiceVersion
-	var err error
+func fetchCSV(c versioned.Interface, name, namespace string, checker csvConditionChecker) (fetched *operatorsv1alpha1.ClusterServiceVersion, err error) {
+	var lastMessage string
 
 	ctx.Ctx().Logf("waiting for CSV %s/%s to reach condition", namespace, name)
-	Eventually(func() (bool, error) {
-		fetchedCSV, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	Eventually(func() bool {
+		fetched, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
-			return false, err
+			ctx.Ctx().Logf("error getting csv %s/%s: %v", namespace, name, err)
+			return false
 		}
-		return checker(fetchedCSV), nil
-	}).Should(BeTrue())
+		msg := fmt.Sprintf("%s/%s - %s (%s): %s", namespace, name, fetched.Status.Phase, fetched.Status.Reason, fetched.Status.Message)
+		if msg != lastMessage {
+			lastMessage = msg
+			ctx.Ctx().Logf(msg)
+		}
+		return checker(fetched)
+	}).WithTimeout(pollDuration).WithPolling(pollInterval).Should(BeTrue())
 
-	if err != nil {
-		ctx.Ctx().Logf("CSV %s/%s never got correct status: %#v", namespace, name, fetchedCSV.Status)
-	}
-	return fetchedCSV, err
+	// Eventually() will error exit on timeout, this only happens on success
+	return fetched, nil
 }
 
-func awaitCSV(c versioned.Interface, namespace, name string, checker csvConditionChecker) (*operatorsv1alpha1.ClusterServiceVersion, error) {
-	var fetched *operatorsv1alpha1.ClusterServiceVersion
-	var err error
-
-	Eventually(func() (bool, error) {
-		fetched, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+func waitForDeployment(namespace string, c operatorclient.ClientInterface, name string) (err error) {
```

Comment:
what's the utility of this func that does not return the delpoyment or wait for any particular condition?

Reply:
This was a preexisting function; it's just waiting for the deployment to be created.

Reply:
Added a return argument, even though it's not used.